core/src/test/scala/org/broadinstitute/dsde/rawls/entities/EntityServiceCompactMigrationSpec.scala

```diff
@@ -414,4 +419,34 @@ class EntityServiceCompactMigrationSpec
     // numAttributesDeleted: we delete all attributes in the workspace, not just the soft-deleted ones
   }
 
+  it should s"roll back the CompactDataTables setting on failed migration" in withTestDataServices { apiService =>
+    val workspace = legacyTestData.workspace // has some entities we can use to test references
+
+    /** Locks all entities of the given workspace for ${lockSeconds} seconds. Use this to simulate database contention. */
+    def lockAllEntities(dataSource: SlickDataSource, workspaceId: UUID, lockSeconds: Int): Future[Unit] = {
+      import dataSource.dataAccess.driver.api._
+      val locker = for {
+        _ <- sql"""select * from ENTITY where workspace_id = $workspaceId for update;""".as[Unit]
+        _ <- sql"""select sleep(${lockSeconds + 2});""".as[Unit]
+      } yield ()
+      dataSource.database.run(locker.transactionally.withTransactionIsolation(TransactionIsolation.Serializable))
+    }
+
+    // lock entities; do NOT wait for this Future to complete so it is still running when we kick off the migration
+    lockAllEntities(slickDataSource, workspace.workspaceIdAsUUID, 60)
+    // ask to perform migration
+    val migrationResult =
+      Try(Await.result(apiService.entityService.quicksilverMigration(workspace.toWorkspaceName), atMost))
+
+    migrationResult shouldBe a[Failure[_]]
+
+    // check if the setting exists
+    val repo = new WorkspaceSettingRepository(slickDataSource)
+    Await.result(repo.hasPendingSettings(workspace.workspaceIdAsUUID, CompactDataTables), Duration.Inf) shouldBe false
+    Await.result(repo.getWorkspaceSettingOfType(workspace.workspaceIdAsUUID, CompactDataTables),
+                 Duration.Inf
+    ) shouldBe empty
+
+  }
+
```

Comment:
This unit test *requires* > 50 seconds to execute - it has to wait for the lock-wait timeout to pass (default is 50 seconds). Is it worth it to add a known-long unit test?

Reply:
Does rawls have a long-running test tag?  I think we have those in wds

Reply:
Maybe once all workspaces are migrated, we can remove this test so it's only a time sink for a little while?